pkg/splunk/enterprise/configuration.go

```diff
@@ -246,27 +246,27 @@ func prepareSplunkSmartstoreConfigMap(identifier, namespace string, crKind strin
 // getSplunkPorts returns a map of ports to use for Splunk instances.
 func getSplunkPorts(instanceType InstanceType) map[string]int {
 	result := map[string]int{
-		"splunkweb": 8000,
-		"splunkd":   8089,
+		GetPortName(splunkwebPort, "http"): 8000,
+		GetPortName(splunkdPort, "https"):  8089,
 	}
 
 	switch instanceType {
 	case SplunkMonitoringConsole:
-		result["hec"] = 8088
-		result["s2s"] = 9997
+		result[GetPortName(hecPort, "http")] = 8088
+		result[GetPortName(s2sPort, "tcp")] = 9997
 	case SplunkStandalone:
-		result["dfccontrol"] = 17000
-		result["datareceive"] = 19000
-		result["dfsmaster"] = 9000
-		result["hec"] = 8088
-		result["s2s"] = 9997
+		result[GetPortName(dfccontrolPort, "tcp")] = 17000
+		result[GetPortName(datareceivePort, "tcp")] = 19000
+		result[GetPortName(dfsmasterPort, "tcp")] = 9000
+		result[GetPortName(hecPort, "http")] = 8088
+		result[GetPortName(s2sPort, "tcp")] = 9997
 	case SplunkSearchHead:
-		result["dfccontrol"] = 17000
-		result["datareceive"] = 19000
-		result["dfsmaster"] = 9000
+		result[GetPortName(dfccontrolPort, "tcp")] = 17000
+		result[GetPortName(datareceivePort, "tcp")] = 19000
+		result[GetPortName(dfsmasterPort, "tcp")] = 9000
 	case SplunkIndexer:
-		result["hec"] = 8088
-		result["s2s"] = 9997
+		result[GetPortName(hecPort, "http")] = 8088
+		result[GetPortName(s2sPort, "tcp")] = 9997
```

Comment:
@vebken-splunk The change looks great. 

Nit: Does it make sense to create string constants for the protocol strings as well?

Reply:
Yeah I guess it's probably worth having them for consistency and potential future usage by other modules. Will update that.